packages/amplify_authenticator/lib/src/mixins/username_field.dart

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+import 'package:amplify_authenticator/src/theme/amplify_theme.dart';
+import 'package:amplify_authenticator/src/widgets/form_field.dart';
+import 'package:flutter/material.dart';
+
+enum _UsernameType {
+  username,
+  email,
+  phoneNumber,
+  emailOrPhoneNumber,
+}
+
+mixin AuthenticatorUsernameField<FieldType,
```

Comment:
Ideally we would make this a generic switch mixin which can be used for other fields. It's not a priority though.